File: Ch06/Reminders/app/src/main/java/com/apress/gerber/reminders/RemindersSimpleCursorAdapter.java
package com.apress.gerber.reminders;

import android.content.Context;
import android.database.Cursor;
import android.support.v4.widget.SimpleCursorAdapter;
import android.view.View;
import android.view.ViewGroup;

public class RemindersSimpleCursorAdapter extends SimpleCursorAdapter {
    public RemindersSimpleCursorAdapter(Context context, int layout, Cursor c,
          String[] from, int[] to, int flags) {
        super(context, layout, c, from, to, flags);
    }

    // ViewHolder 객체`를 사용하려면 다음 두 개의 메서드를 오버라이딩하고
    // ViewHolder 클래스를 정의해야 한다.
    @Override
    public View newView(Context context, Cursor cursor, ViewGroup parent) {
        return super.newView(context, cursor, parent);
    }

    @Override
    public void bindView(View view, Context context, Cursor cursor) {
        super.bindView(view, context, cursor);

        ViewHolder holder = (ViewHolder) view.getTag();
        if (holder == null) {
            holder = new ViewHolder();
            holder.colImp = cursor.getColumnIndexOrThrow(RemindersDbAdapter.COL_IMPORTANT);
            holder.listTab = view.findViewById(R.id.row_tab);
            view.setTag(holder);
        }

        if (cursor.getInt(holder.colImp) > 0) {
            holder.listTab.setBackgroundColor(context.getResources().getColor(R.color.orange));
        } else {
            holder.listTab.setBackgroundColor(context.getResources().getColor(R.color.green));
        }
    }

    static class ViewHolder {
        // 칼럼 인덱스를 저장한다
        int colImp;
        // 뷰를 저장한다
        View listTab;
    }
}

File: Ch06/Reminders/app/src/main/java/com/apress/gerber/reminders/adapter/RemindersSimpleCursorAdapter.kt
package com.apress.gerber.reminders.adapter

import android.content.Context
import android.database.Cursor
import android.view.View
import android.view.ViewGroup
import androidx.cursoradapter.widget.SimpleCursorAdapter
import com.apress.gerber.reminders.R

class RemindersSimpleCursorAdapter(
    context: Context?, layout: Int, c: Cursor?,
    from: Array<String?>?, to: IntArray?, flags: Int
) : SimpleCursorAdapter(context, layout, c, from, to, flags) {
    override fun newView(context: Context, cursor: Cursor, parent: ViewGroup): View {
        return super.newView(context, cursor, parent)
    }

    override fun bindView(view: View, context: Context, cursor: Cursor) {
        super.bindView(view, context, cursor)
        var holder = view.tag
        if (holder == null) {
            holder = ViewHolder()
            holder.colImp = cursor.getColumnIndexOrThrow(COL_IMPORTANT)
            holder.listTab = view.findViewById(R.id.row_tab)
            view.tag = holder
        }
        holder as ViewHolder
        if (cursor.getInt(holder.colImp) > 0) {
            holder.listTab!!.setBackgroundColor(context.resources.getColor(R.color.orange))
        } else {
            holder.listTab!!.setBackgroundColor(context.resources.getColor(R.color.green))
        }
    }

    internal class ViewHolder {
        // 칼럼 인덱스를 저장한다
        var colImp = 0

        // 뷰를 저장한다
        var listTab: View? = null
    }

    companion object {
        const val COL_CONTENT = "content"
        private const val COL_IMPORTANT = "important"
    }
}